### ISSUE TITLE
Unable to use until(Runnable) with Clousure if awaitility-groovy is on classpath

### ISSUE THREAD

**Issue by szpak** [awaitility/awaitility#65]
With awaitility-groovy module on a classpath it is not possible to use `unitl(Runnable)` with Groovy Closure, e.g.

```
await().until({ assertThat(foo).isEqualTo(bar) })
//or
await().until({ wiremock.verifyThat(postRequestedFor(urlEqualTo("...")) }

```

Where `verifyThat` throws AssertError if condition is not fulfilled (the same as AssertJ). This is caused by the fact that AwaitilityExtensionModule treats all `Clusures` as `Runnable`:

```
    if (runnable instanceof Callable)
      self.until(runnable as Callable)
    else
      self.until(new AssertionCondition(runnable, self.generateConditionSettings()))
```

In general it is good, because Groovy by default would use `until(Runnable)` instead of `until(Callable)`. Nevertheless the use case mentioned above is broken.

Unfortunately I don't see a good way to fix it. All ideas I have are far from being perfect:

1. Use `new Runnable() { void run() { ... } }` explicitly.
2. Add marker interface `AwaitilityGroovySupplier` to mark `Closure`s which should be treated as `Runnable` and use it in `AwaitilityExtensionModule`
3. Add another method with different name just for Groovy users
4. Document this limitation in `until(Runnable)` JavaDoc to give people a chance to get know about that.

Maybe you will have some other ideas.

One more thing. Probably not much people use aforementioned path as AssertJ with Groovy is usually not needed and Wiremock is also only for specific scenarios. In addition people using Spock can as an alternative to Awaitility use `PollingCondition` from Spock. In that case maybe the 4rd point would be enough?

**Comment by johanhaleby** [comment]
Thanks for your detailed description and your insights, I didn't know about it and I have to think about it a bit. If you get any new ideas on how to solve it (other than documentation) please share it.

**Comment by johanhaleby** [comment]
It's the same problem when using Kotlin.. Not sure what the best option is to resolve this.

**Comment by johanhaleby** [comment]
I'm thinking of renaming the `Runnable` method to `untilAsserted`. Do you have any thoughts on that?

**Comment by johanhaleby** [comment]
I would be very grateful if you could try out the newly released 3.0.0-rc1 version and give me some feedback of the changes (especially the `untilAsserted` change).

Another idea is to create a stand-alone API for Groovy like @iggymoran is suggesting for Kotlin [here](https://github.com/awaitility/awaitility/issues/81#issuecomment-287455198). Wdyt?

**Comment by szpak** [comment]
I've missed your questions :(. `untilAsserted()` sounds good to me to avoid confusion.
